corehq/apps/app_manager/views/releases.py

```diff
@@ -250,3 +257,100 @@ def update_build_comment(request, domain, app_id):
     build.build_comment = request.POST.get('comment')
     build.save()
     return json_response({'status': 'success'})
+
+
+def _get_app_diff_files(app):
+    """
+    Return a dict of the files that an app build is comprised of. Return dict
+    also includes the app json.
+    """
+    files = dict(download_index_files(app))
+    files["app.json"] = json.dumps(
+        app.to_json(), sort_keys=True, indent=4, separators=(',', ': ')
+    )
+    return files
+
+
+def _get_change_counts(html_diff):
+    diff_lines = html_diff.splitlines()
+    additions = sum(1 for line in diff_lines if line.startswith('<div class="insert">'))
+    deletions = sum(1 for line in diff_lines if line.startswith('<div class="delete">'))
+    return additions, deletions
+
+
+def _get_file_pairs(first_app, second_app):
+    """
+    :param first_app:
+    :param second_app:
+    :return: A dictionary mapping file name to tuple where the first element is
+     the corresponding file on the first app and the second is the corresponding
+     file on the second app.
+     "files" will be empty strings if the file does not exist on the app.
+    """
+    first_app_files = _get_app_diff_files(first_app)
+    second_app_files = _get_app_diff_files(second_app)
+    file_names = set(first_app_files.keys()) | set(second_app_files.keys())
+    file_pairs = {
+        n: (first_app_files.get(n, ""), second_app_files.get(n, ""))
+        for n in file_names
+    }
+    return file_pairs
+
+
+def _get_app_diffs(first_app, second_app):
+    """
+    Return a list of tuples. The first value in each tuple is a file name,
+    the second value is an html snippet representing the diff of that file
+    in the two given apps.
+    """
+    file_pairs = _get_file_pairs(first_app, second_app)
+    diffs = []
+    for name, files in file_pairs.iteritems():
+        diff_html = ghdiff.diff(files[0], files[1], n=4, css=False)
+        additions, deletions = _get_change_counts(diff_html)
+        if additions == 0 and deletions == 0:
+            diff_html = ""
+        diffs.append((name, diff_html, additions, deletions))
+    return sorted(diffs)
```

Comment:
I second @esoergel s comment, and I like that not only are they small and self-explanatory, but also functional. If it wasn't for the fact that they need two versions of an actual app, which would need some work before you can mock in one line, they'd be great candidates for a short doctest each.